lib/google_maps_geocoder.rb

```diff
@@ -59,7 +61,14 @@ def partial_match?
   private
 
   def json_from_url url
-    response = Net::HTTP.get_response(URI.parse "http://maps.googleapis.com/maps/api/geocode/json?address=#{Rack::Utils.escape(url)}&sensor=false")
+    api_key = "&key=#{ENV['GOOGLE_API_TOKEN']}" if ENV['GOOGLE_API_TOKEN']
+    uri = URI.parse "#{GOOGLE_API_URI}?address=#{Rack::Utils.escape(url)}&sensor=false#{api_key}"
+    logger = Logger.new STDERR
+    logger.info('GoogleMapsGeocoder') { "URI: \"#{uri}\"" }
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+    response = http.request(Net::HTTP::Get.new(uri.request_uri))
     ActiveSupport::JSON.decode response.body
```

Comment:
Could you both simplify and optimize this method by:

1. Extracting methods for `api_key`, `logger`, and `http`/`response`.

2. Caching instance variables that can be reused (such as `api_key` and `logger`) in those methods like so...

```ruby
@logger ||= Logger.new STDERR
```

These changes would make the code easier to read and perhaps even (marginally) improve performance.